step-framework-timeseries/src/main/java/step/core/timeseries/ingestion/TimeSeriesIngestionPipeline.java

```diff
@@ -53,7 +53,13 @@ public TimeSeriesIngestionPipeline(TimeSeriesCollection collection, TimeSeriesIn
         this.ignoredAttributes = settings.getIgnoredAttributes();
         this.nextPipeline = settings.getNextPipeline();
         //collection is null when overridden in TimeSeriesExecutionPlugin, in such case async processor is not required
-        this.asyncProcessor =  (collection != null)  ? new AsyncProcessor<>(settings.getFlushAsyncQueueSize(), collection::save) : null;
+        this.asyncProcessor =  (collection != null)  ? new AsyncProcessor<>(settings.getFlushAsyncQueueSize(), entity -> {
```

Comment:
Consider reversing the order here: (collection == null) null : new ...

This makes the `null` case immediately clear, and it avoids that `: null` code a few lines below, where it looks weirdly out of context.